### PR TITLE
chore: add new GH policy requirements

### PR DESCRIPTION
**Pull request by Luisfc68**
## What
- Added dependabot config
- Added dependency review action
- Added scorecard action
- Replace tags by commit hashes in actions
- Added pre commit
- Added contributing, security and license files

## Why
In order to be compliant with the organization's GH policy

## Task
https://rsklabs.atlassian.net/browse/GBI-2134

## Important
The BBP for this repo is not live yet, I think we should wait until it is to merge this PR

### CODE REVIEW

**Review by MaximStanciu8 — APPROVED** [review]

**Review by rafaiovlabs — APPROVED** [review]

**Review by MaximStanciu8 — APPROVED** [review]

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (diego-jeronymo) [inline .github/workflows/dependency-review.yml:20]: suggestion: 

`uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # 4.2.0`

File: .github/workflows/dependency-review.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@72eb03d02c7872a771aacd928f3123ac62ad6d3a # v4.3.3
```
Comment (diego-jeronymo) [inline .github/workflows/dependency-review.yml:22]: suggestion:

`actions/dependency-review-action@5a2ce3f5b92ee19cbb1541a4984c76d921601d7c # v4.3.4`

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
```
Comment (diego-jeronymo) [inline .github/workflows/scorecard.yml:26]: I suggest to use the updated version 2.4.0 
`uses: ossf/scorecard-action@62b2cac7ed8198b15735ed49ab1e5cf35480ba46 # v2.4.0`

_more details: https://github.com/ossf/scorecard-action/releases_

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (diego-jeronymo) [inline .github/workflows/scorecard.yml:21]: suggestion: 

`uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # 4.2.0`

File: .github/workflows/scorecard.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          publish_results: true
+
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@97a0fba1372883ab732affbe8f94b823f91727db # v3.pre.node20
```
Comment (diego-jeronymo) [inline .github/workflows/scorecard.yml:33]: I suggest to use the last updated stable version 2.4.0
`uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0`
_more details: https://github.com/actions/upload-artifact/releases_

File: .github/workflows/scorecard.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
+
+      - name: "Upload to code-scanning"
+        uses: github/codeql-action/upload-sarif@1b1aada464948af03b950897e5eb522f92603cc2 # v3.24.9
```
Comment (diego-jeronymo) [inline .github/workflows/scorecard.yml:40]: I suggest to use the last updated stable version 3.26.9
`uses: github/codeql-action/upload-sarif@461ef6c76dfe95d5c364de2f431ddbd31a417628 # v3.26.9`

File: .github/workflows/slither.yml [hunk 7]
```diff
@@ -1,30 +1,34 @@
...
+  contents: read
+
 jobs:
   analyze:
     runs-on: ubuntu-latest
     permissions:
       security-events: write
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (diego-jeronymo) [inline .github/workflows/slither.yml:13]: suggestion: 

`uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # 4.2.0`

File: .github/workflows/slither.yml [hunk 8]
```diff
@@ -1,30 +1,34 @@
...
+        uses: crytic/slither-action@6ef3a33e56de4e8f59488cf60858b5c1bf4967c0 # v0.3.0
         id: slither
         with:
           sarif: results.sarif
           fail-on: none
           target: .
 
       - name: Upload SARIF file
-        uses: github/codeql-action/upload-sarif@v2
+        uses: github/codeql-action/upload-sarif@85b07cf1e13dd512be7c27c37a33c5864c252fcc # v2
```
Comment (diego-jeronymo) [inline .github/workflows/slither.yml:32]: I suggest to use the last updated stable version 3.26.9
`uses: github/codeql-action/upload-sarif@461ef6c76dfe95d5c364de2f431ddbd31a417628 # v3.26.9`

**Pull request merged**